repo/fsrepo/fsrepo.go

```diff
@@ -562,6 +563,22 @@ func (r *FSRepo) Datastore() ds.ThreadSafeDatastore {
 	return d
 }
 
+// GetDiskUsage computes the disk space taken by the repo in bytes
+func (r *FSRepo) GetDiskUsage() (int64, error) {
+	pth, err := config.PathRoot()
+	if err != nil {
+		return 0, err
+	}
+
+	var du int64
+	err = filepath.Walk(pth, func(p string, f os.FileInfo, err error) error {
+		fi, err := os.Stat(p)
+		du += fi.Size()
```

Comment:
if os.Stat errors, fi might be nil, check and return early in that case. and maybe debug log the error